### PR TITLE
Create the History section of JP Scan

### PR DESCRIPTION
**Pull request by rcanepa**
**Changes proposed in this Pull Request**

- The goal of this is to create the iteration of the UI of the History section. At this point, we don't know if we will have an APIs ready to work with.

See 1151678672052943-as-1164658985934865.

**Testing instructions**

- View Scan page and verify that component matches comps.

**Screenshots**
Kapture 2020-03-05 at 22 58 33

**Desktop**
image

**Mobile**
image

### CODE REVIEW

File: client/landing/jetpack-cloud/sections/scan/history/index.jsx [hunk 1]
```diff
@@ -1,17 +1,159 @@
...
+		entry: PropTypes.object,
+	};
+
+	renderEntryHeader() {
+		const { entry } = this.props;
+
+		const actionClass = 'entry' + entry.action === 'fixed' ? ' is-fixed' : ' is-ignored';
+
+		return (
+			<div className={ actionClass }>
```
Comment (enejb) [inline client/landing/jetpack-cloud/sections/scan/history/index.jsx]: Can we use the <LogItem component that is found in `landing/jetpack-cloud/components` instead of creating our own?

Comment (rcanepa) [reply client/landing/jetpack-cloud/sections/scan/history/index.jsx]: Sure! The only thing that makes me uncomfortable is that this component has a `highlight` property to set its color. The options are 'info', 'success', 'warning', and 'error', and neither of those matches the color we need (--studio-jetpack-green-40). Should we use it anyway and override the color from the parent component? Is that what we do in a case like this?

Comment (rcanepa) [reply client/landing/jetpack-cloud/sections/scan/history/index.jsx]: I faced the same question when working on the badge (fixed or ignored just before the title of the entry). There is Badge component, is looks similar to what is in the designs. It also accepts the same `highlight` property. Should I use that o create a new one?

**Comment by enejb** [comment]
Instead of adding a new `className` attribute to LogItem can we use the existing hightlight feature that already exists in the component? 

We need to extend it to implement the 'ignored' case better.

File: client/landing/jetpack-cloud/components/scan-history-item/style.scss [hunk 2]
```diff
@@ -0,0 +1,65 @@
+.scan-history-item {
+	&.is-fixed {
+		border-left: 3px solid var( --studio-jetpack-green-40 );
```
Comment (enejb) [inline client/landing/jetpack-cloud/components/scan-history-item/style.scss]: Instead of using the `--studio-jetpack-green-40` can we use the variables that are now available since https://github.com/Automattic/wp-calypso/pull/39878 got merged.

File: client/landing/jetpack-cloud/sections/scan/history/index.jsx [hunk 3]
```diff
@@ -3,22 +3,162 @@
...
+	},
+];
+
 class ScanHistoryPage extends Component {
+	state = {
+		filter: filterOptions[ 0 ],
+	};
+
+	handleOnFilterChange = filter => {
+		// @todo: should we filter in the front end?
```
Comment (enejb) [inline client/landing/jetpack-cloud/sections/scan/history/index.jsx]: This look great. 
I think we should also add the filter to the url as the user navigates switches between the filters.
Lets do this in a different PR. 

We should also add analytics tracking to this event. This should also be done in a different PR.

**Comment by enejb** [comment]
Nice work @rcanepa! 

I left some comments on the PR. I think we are getting very close to merging it.

File: client/landing/jetpack-cloud/components/log-item/index.tsx [hunk 4]
```diff
@@ -35,9 +37,13 @@ class LogItem extends React.PureComponent< Props > {
 	}
 
 	render() {
-		const { highlight, children } = this.props;
+		const { highlight, children, className } = this.props;
 		return (
-			<FoldableCard header={ this.renderHeader() } className="log-item" highlight={ highlight }>
+			<FoldableCard
```
Comment (ChaosExAnima) [inline client/landing/jetpack-cloud/components/log-item/index.tsx]: Out of curiosity, could you use the `ActivityList` for items like this?

File: client/landing/jetpack-cloud/components/scan-history-item/index.jsx [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+	renderEntryHeader( entry ) {
+		// React complains because we shouldn't wrap a <div> tag inside <p> tag. This happens
+		// because LogItem surrounds its content by a <p> tag, and the Badge is a <div> tag.
+		// Should we create our own Badge component?
+		return (
+			<>
+				<div className="scan-history-item__subheader">
+					<span className="scan-history-item__date">Threat found on { entry.detectionDate }</span>
+					<span className="scan-history-item__date-separator"></span>
+					<span className={ classnames( 'scan-history-item__date', this.entryActionClassNames() ) }>
```
Comment (robertf4) [inline client/landing/jetpack-cloud/components/scan-history-item/index.jsx]: This line is causing a crash for me because the call to `this.entryActionClassNames()` doesn't pass anything for the `entry` parameter and so there is a null reference error.

File: client/landing/jetpack-cloud/components/scan-history-item/style.scss (hunk 2 above)
Comment (rcanepa) [reply client/landing/jetpack-cloud/components/scan-history-item/style.scss]: Sure we can!

File: client/landing/jetpack-cloud/components/scan-history-item/index.jsx [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+		return {
+			'is-fixed': entry.action === 'fixed',
+			'is-ignored': entry.action === 'ignored',
+		};
+	}
+
+	renderEntryHeader( entry ) {
+		// React complains because we shouldn't wrap a <div> tag inside <p> tag. This happens
+		// because LogItem surrounds its content by a <p> tag, and the Badge is a <div> tag.
+		// Should we create our own Badge component?
```
Comment (robertf4) [inline client/landing/jetpack-cloud/components/scan-history-item/index.jsx]: > Should we create our own Badge component?

Since `LogItem` is a Jetpack component, I think it is better to change that `<p>` tag to a `<div>`. The component's interface is typed such that `subheader` is `string | ReactNode`, and since `subheader` can be a `ReactNode` then it probably shouldn't be wrapped in a `<p>`.

File: client/landing/jetpack-cloud/components/scan-history-item/index.jsx [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+		return (
+			<>
+				<div className="scan-history-item__subheader">
+					<span className="scan-history-item__date">Threat found on { entry.detectionDate }</span>
```
Comment (robertf4) [inline client/landing/jetpack-cloud/components/scan-history-item/index.jsx]: The string here needs to be passed through `localize`. See [here](https://github.com/Automattic/wp-calypso/tree/master/packages/i18n-calypso#translate-method) for detailed usage.

File: client/landing/jetpack-cloud/components/scan-history-item/index.jsx [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+					<span className="scan-history-item__date-separator"></span>
+					<span className={ classnames( 'scan-history-item__date', this.entryActionClassNames() ) }>
+						Threat { entry.action } on { entry.actionDate }
```
Comment (robertf4) [inline client/landing/jetpack-cloud/components/scan-history-item/index.jsx]: This string also needs to be passed to `localize()`

File: client/landing/jetpack-cloud/sections/scan/history/style.scss [hunk 9]
```diff
@@ -0,0 +1,16 @@
+.history {
+	padding: 0 10px;
+
+	&__title {
+		text-align: left;
+		margin-left: -10px;
```
Comment (robertf4) [inline client/landing/jetpack-cloud/sections/scan/history/style.scss]: It looks like the history title might be aligned too far left:

image

**Comment by rcanepa** [comment]
> LogItem

**Comment by rcanepa** [comment]
> Instead of adding a new `className` attribute to LogItem can we use the existing hightlight feature that already exists in the component?
> 
> We need to extend it to implement the 'ignored' case better.

Not just the `'ignored'` case, we would have to add `'fixed'` as well because the `'fixed'` case doesn't use the same color as the `'success'` case. They are different type of greens.

I investigated this option but it's not that easy. LogItem renders a FoldableCard and the FoldableCard renders a Card or a CompactCard as well. Therefore, to make the highlight property accepts those two values we would need to touch all jsx and scss files associated with those components. This increases the risk associated with this change because we would need to modify Calypso components.

Also, the highlight property supports the following list of values: `'info'`, `'success'`, `'error'`, and `'warning'`. I believe most developers are familiar with those values because they are widely used to style notification/info components, while `'fixed'` and '`ignored'` don't seem to belong to the same group.

Maybe we should address this in another PR. Something very similar is required to build the "Threat Card". What do you think?

File: client/landing/jetpack-cloud/sections/scan/history/index.jsx (hunk 3 above)
Comment (rcanepa) [reply client/landing/jetpack-cloud/sections/scan/history/index.jsx]: > We should also add analytics tracking to this event.

How do we track events?

File: client/landing/jetpack-cloud/components/log-item/index.tsx (hunk 4 above)
Comment (rcanepa) [reply client/landing/jetpack-cloud/components/log-item/index.tsx]: Do you mean to use `ActivityList` to wrap the list of ignored and fixed events? In theory, we could but I'm not convinced we should. The UIs of the backup and scan events are not the same, so we would have to add to the `ActivityList` non-trivial logic to work well in both cases.

File: client/landing/jetpack-cloud/components/scan-history-item/index.jsx (hunk 5 above)
Comment (rcanepa) [reply client/landing/jetpack-cloud/components/scan-history-item/index.jsx]: Good catch!

File: client/landing/jetpack-cloud/components/scan-history-item/index.jsx (hunk 6 above)
Comment (rcanepa) [reply client/landing/jetpack-cloud/components/scan-history-item/index.jsx]: This is an awesome suggestion! Thank you!

File: client/landing/jetpack-cloud/components/scan-history-item/index.jsx (hunk 7 above)
Comment (rcanepa) [reply client/landing/jetpack-cloud/components/scan-history-item/index.jsx]: Done!

File: client/landing/jetpack-cloud/components/scan-history-item/index.jsx (hunk 8 above)
Comment (rcanepa) [reply client/landing/jetpack-cloud/components/scan-history-item/index.jsx]: Done!

File: client/landing/jetpack-cloud/sections/scan/history/style.scss (hunk 9 above)
Comment (rcanepa) [reply client/landing/jetpack-cloud/sections/scan/history/style.scss]: I changed the `FormattedHeader` for an `h1` and give it a specific font-size depending on the device: mobile or desktop.

File: client/landing/jetpack-cloud/sections/scan/history/index.jsx (hunk 1 above)
Comment (ChaosExAnima) [reply client/landing/jetpack-cloud/sections/scan/history/index.jsx]: I wrote that component when the colors weren't finalized. Feel free to adjust them!

**Comment by enejb** [comment]
> Maybe we should address this in another PR. Something very similar is required to build the "Threat Card". What do you think?

Lets think about this some more and addressed in a similar way on the

**Comment by keoshi** [comment]
This looks incredible! Kudos @rcanepa
